src/windows-emulator/memory_manager.hpp

```diff
@@ -12,13 +12,22 @@ constexpr auto ALLOCATION_GRANULARITY = 0x0000000000010000ULL;
 constexpr auto MIN_ALLOCATION_ADDRESS = 0x0000000000010000ULL;
 constexpr auto MAX_ALLOCATION_ADDRESS = 0x00007ffffffeffffULL;
 
-struct region_info : basic_memory_region
+// This maps to the `basic_memory_region` struct defined in
+// emulator\memory_region.hpp
+struct nt_basic_memory_region
+{
+    uint64_t start{};
+    size_t length{};
+    nt_memory_permission permissions{};
+};
+
+struct region_info : nt_basic_memory_region
```

Comment:
i would recommend making the basic_memory_region a template, something like this:

```C++
template<typename PermissionType = memory_permission>
struct basic_memory_region
{
    uint64_t start{};
    size_t length{}; // uint64_t?
    PermissionType permissions{};
};
```

that way, region_info can inherit from that with a custom permission type:
```suggestion
struct region_info : basic_memory_region<nt_memory_permission>
```